controller/src/controller/utils/serial_comm.py

```diff
@@ -492,3 +499,33 @@ def convert_stim_bytes_to_dict(stim_bytes: bytes) -> dict[str, Any]:
         curr_byte_idx += num_wells_assigned
 
     return stim_info_dict
+
+
+def parse_end_offline_mode_bytes(response_bytes: bytes) -> dict[str, Any]:
+    """Parse bytes containing stimulation info and return as Dict."""
+
+    stim_dict = convert_stim_bytes_to_dict(response_bytes[41:])
+    updated_stim_dict = format_stim_dict_with_ids(stim_dict)
+
+    return {
+        "system_dormant_timestamp": int.from_bytes(response_bytes[:8]),
+        "stim_active": bool(response_bytes[8]),
+        "last_stim_scheduled_start_timestamp": int.from_bytes(response_bytes[9:17]),
+        "stimulator_statuses": response_bytes[17:41],  # TODO will need to update with final dict here
+        "stim_info": updated_stim_dict,
+    }
+
+
+def format_stim_dict_with_ids(stim_dict: dict[str, Any]) -> dict[str, Any]:
+    """Add protocol IDs to stim dict from instrument after being offline."""
+    for idx, _ in enumerate(stim_dict["protocols"]):
+        protocol_id = chr(idx + 97).upper()
```

Comment:
It's a little less clean but more clear IMO to do `chr(idx + old("A"))`

Reply:
Also minor but I don't think `upper()` is needed here
```
>>> [chr(idx + ord("A")) for idx in range(5)]
['A', 'B', 'C', 'D', 'E']
```

Reply:
`upper()` is required when doing it the way I had, not with `chr(idx + ord("A"))`

Reply:
Because 97 is a magic number in this case, especially if someone isn't familiar with ASCII values or ASCII/chars in general. Using `ord("A")` just makes it clear that we're starting from the int value of `'A'`